apps/meteor/app/lib/server/functions/closeOmnichannelConversations.ts

```diff
@@ -3,18 +3,22 @@ import type { IUser } from '@rocket.chat/core-typings';
 
 import { LivechatRooms } from '../../../models/server';
 import { settings } from '../../../settings/server';
-import { Livechat } from '../../../livechat/server/lib/Livechat';
+import { Livechat } from '../../../livechat/server/lib/LivechatTyped';
 
 type SubscribedRooms = {
 	rid: string;
 	t: string;
 };
 
-export const closeOmnichannelConversations = (user: IUser, subscribedRooms: SubscribedRooms[]): void => {
+export const closeOmnichannelConversations = async (user: IUser, subscribedRooms: SubscribedRooms[]): Promise<void> => {
 	const roomsInfo = LivechatRooms.findByIds(subscribedRooms.map(({ rid }) => rid));
 	const language = settings.get<string>('Language') || 'en';
 	const comment = TAPi18n.__('Agent_deactivated', { lng: language });
+
+	const promises: Promise<void>[] = [];
 	roomsInfo.forEach((room: any) => {
-		Livechat.closeRoom({ user, visitor: {}, room, comment });
+		promises.push(Livechat.closeRoom({ user, room, comment }));
 	});
```

Comment:
@KevLehman

Reply:
@aleksandernsilva sadly that cannot be done, since `roomsInfo` is a mongo cursor, and it only has forEach before converting it to array. So we cannot use array methods on it :sadblob: